Source/UI/NewUI/PaymentForm/View/CardNumber/CardNumberViewModel.swift

```diff
@@ -32,6 +40,8 @@ class CardNumberViewModel {
 }
 
 extension CardNumberViewModel: CardNumberViewModelProtocol {
+
+  // TODO: separate the validation logic from getting the icon
```

Comment:
Do we need this TODO? Why do we need to separate that logic, if the validation ties this logic already?

I validate a card so now I know if it is of any known scheme and whether its a complete valid entry for it. Its all 1 validation, not sure how you'd like to separate?

Reply:
`func validate(cardNumber rawText: String) -> Constants.Bundle.SchemeIcon?` 
function is asking for validation and return image. 
`validate` is expected return boolean then with the boolean get image or locatization or or anything.

Reply:
@ehab-al-cko , I strongly disagree with your understanding of what validation does. It obviously is not a question of:
"Is this valid", but more of "What type is it". Please try to understand the wider picture clearer. 

I will push back against such a refactor here as it just does not make sense. API methods should not only be expected to return booleans, they are perfectly fine returning other types that can then be used to infer your needs.

As a result to move forward, please remove the TODO and open a Jira task that we can scope and discuss on.